source/ngf-mtl/impl.cpp

```diff
@@ -1033,6 +1035,244 @@ static void ngfmtl_populate_ngf_device(uint32_t handle, ngf_device& ngfdev, MTL:
 
 NGFI_THREADLOCAL ngf_context CURRENT_CONTEXT = nullptr;
 
+static ngf_error ngf_resolve_counter_timestamp(
```

Comment:
5. 
almost all of the remaining code here (finding the necessary counter set, creating counter sample buffer etc) can all be moved up into the application layer: you can do all that stuff in the app (using the device handle provided by `ngf_get_mtl_device`), and as long as you provide nicegraf with your sample buffer attachment descriptors prior to starting a pass, everything should work.